packages/make-styles/src/runtime/resolveStyleRules.test.ts

```diff
@@ -0,0 +1,175 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore No typings :(
+import * as prettier from 'prettier';
+
+import { resolveStyleRules } from './resolveStyleRules';
+import { MakeStylesResolvedRule } from '../types';
+import { isObject } from './utils/isObject';
+
+expect.addSnapshotSerializer({
+  test(value) {
+    return isObject(value);
+  },
+  print(value: Record<string, MakeStylesResolvedRule>) {
+    return Object.keys(value).reduce((acc, property) => {
+      const rule: MakeStylesResolvedRule = value[property];
+
+      return prettier.format(acc + rule[1] + (rule[2] || ''), { parser: 'css' }).trim();
+    }, '');
+  },
+});
+
+describe('resolveStyleRules', () => {
+  describe('css', () => {
+    it('resolves a single rule', () => {
+      expect(resolveStyleRules({ color: 'red' })).toMatchInlineSnapshot(`
+        .fe3e8s9 {
+          color: red;
+        }
+      `);
+    });
+
+    it('resolves multiple rules', () => {
+      expect(resolveStyleRules({ backgroundColor: 'green', color: 'red' })).toMatchInlineSnapshot(`
+        .fcnqdeg {
+          background-color: green;
+        }
+        .fe3e8s9 {
+          color: red;
+        }
+      `);
+    });
+
+    it('performs expansion of shorthands', () => {
+      expect(resolveStyleRules({ outline: '1px' })).toMatchInlineSnapshot(`
+        .fpvhumw {
+          outline-width: 1px;
+        }
+      `);
+      expect(resolveStyleRules({ padding: '5px' })).toMatchInlineSnapshot(`
+        .f1sbtcvk {
+          padding-top: 5px;
+        }
+        .fwiuce9 {
+          padding-right: 5px;
+        }
+        .rfwiuce9 {
+          padding-left: 5px;
+        }
+        .fdghr9 {
+          padding-bottom: 5px;
+        }
+        .f15vdbe4 {
+          padding-left: 5px;
+        }
+        .rf15vdbe4 {
+          padding-right: 5px;
+        }
+      `);
+    });
+
+    it('performs vendor prefixing', () => {
+      expect(resolveStyleRules({ display: 'flex' })).toMatchInlineSnapshot(`
+        .f22iagw {
+          display: flex;
+        }
+      `);
+    });
+
+    it('handles falsy values', () => {
+      expect(
+        resolveStyleRules({
+          zIndex: 1,
+          position: (null as unknown) as undefined,
+          top: undefined,
+        }),
+      ).toMatchInlineSnapshot(`
+        .f19g0ac {
+          z-index: 1;
+        }
+      `);
+    });
+
+    it('handles RTL', () => {
+      expect(resolveStyleRules({ left: '5px' })).toMatchInlineSnapshot(`
+        .f5b3q4t {
+          left: 5px;
+        }
+        .rf5b3q4t {
+          right: 5px;
+        }
+      `);
+    });
+
+    it('handles media queries', () => {
+      expect(
+        resolveStyleRules({
+          color: 'green',
+          '@media screen and (max-width: 992px)': { color: 'red ' },
```

Comment:
There is an extra space after `red`. When you remove the space the classname (hash) changes. Should we trim the values?

Reply:
Agree, fixed in 181669d